### PR TITLE
[pbjson] Add new port

### LINKED ISSUE #13131: [New Port Request] pbjson
Can you please add https://github.com/yinqiwen/pbjson library?

### PR DESCRIPTION
**Pull request by 0xtejas**
- Fixes #13131 
- src file added

### CODE REVIEW

**Comment by NancyLi1013** [comment]
Hi @tejas15802 
Thanks for adding this port.

It seems that you just add the source code to this PR.

Please refer to this doc https://github.com/microsoft/vcpkg/blob/master/docs/examples/packaging-github-repos.md to learn about how to add a new port. 

You can also search other ports in vcpkg to learn more about this.

BTW, please sign CLA first.

**Comment by tejas15802** [comment]
@NancyLi1013 Than code doesn't owned by me. So I'm little confused on what option I should be choosing for signing

**Comment by tejas15802** [comment]
can you please guide me choose the appropriate option @NancyLi1013

**Comment by NancyLi1013** [comment]
@tejas15802 

You can click here https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=13160 to sign CLA.

**Comment by tejas15802** [comment]
> @tejas15802
> 
> You can click here https://cla.opensource.microsoft.com/microsoft/vcpkg?pullRequest=13160 to sign CLA.

I'm not the author of that code, so I'm not sure If I can sign it

**Comment by NancyLi1013** [comment]
This is not related to the author. So it doesn't matter to sign this. Everyone needs to sign this if he wants to contribute to Microsoft open source projects. Since the source codes have included LICENSE file, you don't need to worry about this.

**Comment by tejas15802** [comment]
I don't what is causing the issue, can someone help me out?

**Comment by NancyLi1013** [comment]
@tejas15802 
You just need to keep CONTROL, portfile.cmake files in this PR. Other files are not needed here. You should remove these files.

BTW, I noticed that this is a header-only library, in this case, you just need to install its headers to vcpkg.
Please refer to this port https://github.com/microsoft/vcpkg/tree/master/ports/libmorton.

**Comment by tejas15802** [comment]
Have I done it right this time, coz it's taking long to verify

File: ports/pbjson/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    REPO yinqiwen/pbjson
+    REF v1.0
+    SHA512 ef84dcd764eb9edfef69f279d020aeff93edf906015c20df0b2528091533d7263221605583a738022db39724ec38f545feca68acc97462a855a7d8d7d461c56a
+    HEAD_REF master
+)
+
+file(COPY ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/pbjson)
+file(RENAME ${CURRENT_PACKAGES_DIR}/share/pbjson/LICENSE ${CURRENT_PACKAGES_DIR}/share/pbjson/copyright)
+
+file(GLOB HEADER_FILES ${SOURCE_PATH}/src/*.h)
```
Comment (NancyLi1013) [inline ports/pbjson/portfile.cmake:15]: It seems that there are also *.hpp files included in the source. So these kinds of headers are also needed to install.

BTW, I also noticed that there are `rapidjson` source codes in _${SOURCE_PATH}/src_ directory. Do you know what the relationships between `pbjson `and `rapidjson`?

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/pbjson/portfile.cmake (hunk 1 above)
Comment (tejas15802) [reply ports/pbjson/portfile.cmake:15]: no I don't know : [

File: ports/pbjson/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO yinqiwen/pbjson
+    REF v1.0
+    SHA512 ef84dcd764eb9edfef69f279d020aeff93edf906015c20df0b2528091533d7263221605583a738022db39724ec38f545feca68acc97462a855a7d8d7d461c56a
+    HEAD_REF master
+)
+
+file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
+
+file(GLOB HEADER_FILES ${SOURCE_PATH}/src/*.h)
```
Comment (NancyLi1013) [inline ports/pbjson/portfile.cmake:13]:
```suggestion
file(GLOB HEADER_FILES ${SOURCE_PATH}/src/*.h  ${SOURCE_PATH}/src/*.hpp)
```
Please also add .hpp file.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Review by 0xtejas — COMMENTED** [review]
Updated the same

**Comment by tejas15802** [comment]
should I look @ the code once again?

**Comment by NancyLi1013** [comment]
> should I look @ the code once again?

Sorry, I don't make sense. What do you mean?

If you mean whether you need to update changes or not in this PR, I think it's unnecessary now.
Since the current status seems to be not related with this PR. I will try to rerun it.

**Comment by tejas15802** [comment]
> > should I look @ the code once again?
> 
> Sorry, I don't make sense. What do you mean?
> 
> If you mean whether you need to update changes or not in this PR, I think it's unnecessary now.
> Since the current status seems to be not related with this PR. I will try to rerun it.

Yes that's what I meant. Mm Okay!

**Comment by NancyLi1013** [comment]
Hi @tejas15802 
Sorry for the long delay.

It seems that the repository doesn't exist now.

Could you please help take a look and merge master changes to this PR?

**Comment by tejas15802** [comment]
I actually deleted the repo, cuz I thought I made some kind of mistake and what I did isn't going to work

**Comment by tejas15802** [comment]
I actually deleted the repo, cuz I thought I made some kind of mistake and what I did isn't going to work 

@NancyLi1013 


> Hi @tejas15802
> Sorry for the long delay.
> 
> It seems that the repository doesn't exist now.
> 
> Could you please help take a look and merge master changes to this PR?

**Comment by NancyLi1013** [comment]
Are you willing to create a new repo to make a new PR for the new port?

Without the original repo, this PR cannot continue working.

**Comment by tejas15802** [comment]
> Are you willing to create a new repo to make a new PR for the new port?
> 
> Without the original repo, this PR cannot continue working.

sorry I'm not.

**Comment by NancyLi1013** [comment]
Thanks for the PR; we're closing this for now since the repo has been deleted. If you'd like to continue working for the port, please open a new PR.

**Pull request closed without merging**